ros2cli/ros2cli/node/daemon.py

```diff
@@ -72,11 +72,14 @@ def spawn_daemon(args, wait_until_spawned=None):
             'Unable to get rmw_implementation_identifier, '
             'try specifying the implementation to use via the '
             "'RMW_IMPLEMENTATION' environment variable")
-    subprocess.Popen(cmd + [
+    cmd.extend([
         # the arguments are only passed for visibility in e.g. the process list
         '--rmw-implementation', rmw_implementation_identifier,
-        '--ros-domain-id', str(ros_domain_id)],
-        stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, **kwargs)
+        '--ros-domain-id', str(ros_domain_id)])
+    if not hasattr(args, 'debug') or not args.debug:
+        kwargs['stdout'] = subprocess.DEVNULL
+        kwargs['stderr'] = subprocess.DEVNULL
+    subprocess.Popen(cmd, **kwargs)
```

Comment:
@ivanpauno meta: I had not noticed, and it predates this PR, but this is a peculiar way to demonize a process.

Reply:
I don't know what is the correct way, but in any case, I would make that modification in a follow-up PR.